evap/evaluation/tests/test_models_logging.py

```diff
@@ -147,3 +147,16 @@ def test_simultaneous_add_and_remove(self):
             self.evaluation.related_logentries().order_by("id").last().data,
             {"participants": {"add": [participant2.id], "remove": [participant1.id]}},
         )
+
+    def test_logging_respects_unlogged_fields(self):
+        participant = baker.make(UserProfile)
+
+        # Temporarily modify unlogged_fields to include 'participants'
+        original_unlogged_fields = self.evaluation.unlogged_fields
+        self.evaluation.__class__.unlogged_fields = property(lambda self: original_unlogged_fields + ["participants"])
+
+        self.evaluation.participants.add(participant)
+        self.assertFalse(any("participants" in entry.data for entry in self.evaluation.related_logentries()))
+
+        # Restore original unlogged_fields or other test might have unexpected behavior
+        self.evaluation.__class__.unlogged_fields = property(lambda self: original_unlogged_fields)
```

Comment:
If the assertion fails (or anything else goes wrong), we never arrive at the code that undoes our hot patching.

Can you use [`unittest.mock.patch`](https://docs.python.org/3/library/unittest.mock.html#unittest.mock.patch) (or something similar) instead, with the context-manager approach, so that we have a guarantee that the modification is undone?